pandas/core/arrays/period.py

```diff
@@ -360,7 +354,12 @@ def _add_offset(self, other):
         if base != self.freq.rule_code:
             msg = DIFFERENT_FREQ_INDEX.format(self.freqstr, other.freqstr)
             raise IncompatibleFrequency(msg)
-        return self._time_shift(other.n)
+
+        # Note: when calling parent class's _add_delta_td, it will call
+        #  delta_to_nanoseconds(delta).  Because delta here is an integer,
+        #  delta_to_nanoseconds will return it unchanged.
+        result = DatetimeLikeArrayMixin._add_delta_td(self, other.n)
```

Comment:
Remind me, why call DatetimeLikeArrayMixin._add_delta_td directly, rather than `super`?

Reply:
I've gotten in the habit because of the complicated inheritance structures in these classes.

Reply:
:) hopefully we can reduce that complexity a bit